src/io/flutter/editor/OutlineLocation.java

```diff
@@ -96,19 +96,24 @@ public void dispose() {
    */
   public void track(Document document) {
     if (marker != null) {
-      // TODO(jacobr): it does indicate a bit of a logic bug if we are calling
-      // this method twice.
+      // TODO(jacobr): it does indicate a bit of a logic bug if we are calling this method twice.
       assert (marker.getDocument() == document);
       return;
     }
+
     assert (indent <= column);
-    assert (marker == null);
-    final int delta = max(column - indent, 0);
+
+    final int delta = Math.max(column - indent, 0);
     final int markerEnd = offset;
     // Create a range marker that goes from the start of the indent for the line
-    // to the column of the actual entity;
-    marker = document.createRangeMarker(max(markerEnd - delta, 0), min(markerEnd + 1, document.getTextLength()));
-    nodeStartingWord = getCurrentWord(document, markerEnd);
+    // to the column of the actual entity.
+    final int startOffset = Math.max(markerEnd - delta, 0);
+    final int endOffset = Math.min(markerEnd + 1, document.getTextLength());
+
```

Comment:
consider instead making
endOffset = max(startOffset, endOffset)
or
flagging the outline location as invalid and avoiding displaying it in the UI guides.
The current option will fix the crash but would have the unintended consequence that we use an untracked location for getColumn and related methods even though `track` has been called. This could result in nonsense UI guides after the user makes a couple local edits.

Reply:
OK, makes sense. I updated this code to make sure that `startOffset` isn't past the end of the file (which I think was the underlying problem w/ the reported bug). And, we don't conditionally create the range marker now.